### PR TITLE
Destroy sessions of stopped service

### PR DESCRIPTION
**Pull request by donce**
Fixes https://github.com/mysteriumnetwork/node/issues/829

### CODE REVIEW

File: core/service/manager_test.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 	"errors"
 	"testing"
 
+	"github.com/asaskevich/EventBus"
```
Comment (Waldz) [inline core/service/manager_test.go]: What kind of dependency is here, I think we have own implementation of event wo external dependencies

File: session/storage_memory.go [hunk 2]
```diff
@@ -81,3 +81,13 @@ func (storage *StorageMemory) Remove(id ID) {
 	defer storage.lock.Unlock()
 	delete(storage.sessions, id)
 }
+
+// RemoveForProposal removes all sessions with given proposal id
+func (storage *StorageMemory) RemoveForProposal(proposalID int) {
```
Comment (Waldz) [inline session/storage_memory.go]: Why filtering by proposalId, it should search by unique index (like `session.id`) and there is a function for that already `storage.Remove(id)`

File: session/manager.go [hunk 3]
```diff
@@ -131,6 +131,7 @@ func (manager *Manager) Create(consumerID identity.Identity, issuerID identity.I
 	sessionInstance.ConsumerID = consumerID
 	sessionInstance.done = make(chan struct{})
 	sessionInstance.Config = config
+	sessionInstance.ProposalID = proposalID
```
Comment (Waldz) [inline session/manager.go]: I don't think we this field in session, it's a serial number for proposal but not unique index itself

File: core/service/manager.go [hunk 4]
```diff
@@ -171,7 +182,18 @@ func (manager *Manager) Kill() error {
...
+	if instance == nil {
+		return errors.New("service not found")
+	}
+
+	err := manager.servicePool.Stop(id)
+	if err != nil {
+		return err
+	}
+
+	manager.eventPublisher.Publish(StopTopic, instance)
```
Comment (Waldz) [inline core/service/manager.go]: I expect all session stop logic to live in `servicePool.Stop()`. No point to do searching twice, it's already done there

File: core/service/manager_test.go (hunk 1 above)
Comment (donce) [reply core/service/manager_test.go]: It doesn't seem that we have our own implementation - we're even using the interface from this external repo in `di.go`:
https://github.com/mysteriumnetwork/node/blob/master/cmd/di.go#L25

File: session/storage_memory.go (hunk 2 above)
Comment (donce) [reply session/storage_memory.go]: When stopping service, we need to find sessions which belong to this service.
I've done this outside of `StorageMemory` at first, but then that method was feature-envy. So I've moved that method here.

File: session/manager.go (hunk 3 above)
Comment (donce) [reply session/manager.go]: It's not supposed to be unique index, we have session id for that. This field allows finding which sessions belong to service being stopped.

File: core/service/manager.go (hunk 4 above)
Comment (donce) [reply core/service/manager.go]: At first I've done all cleanup directly, but then I've added this event as request from @vkuznecovas .
The idea is that all the management is getting too complex, so we want to decouple different components/features by using event bus, signalling all events (service start, stop and possibly others).

At this point, this event doesn't seem very useful, but it's a start towards decoupling :)

File: cmd/service_bootstrap_desktop.go [hunk 5]
```diff
@@ -185,5 +186,11 @@ func (di *Dependencies) bootstrapServiceComponents(nodeOptions node.Options) {
 		newDialogHandler,
 		newDiscovery,
 		di.NATPinger,
+		di.EventBus,
 	)
+
+	serviceCleaner := service.Cleaner{SessionStorage: di.ServiceSessionStorage}
+	if err := di.EventBus.Subscribe(service.StopTopic, serviceCleaner); err != nil {
```
Comment (zolia) [inline cmd/service_bootstrap_desktop.go]: We probably we should call a Cleanup function, not pass an instance of Cleaner

File: core/service/manager.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (donce) [reply core/service/manager.go]: Well, we call it `Topic` everywhere, even though we're using `EventBus` term for the message queue. Actually, this naming doesn't come from us - it comes from asaskevich/event-bus repo, which we are coupled to somehow.

I've created a separate task to decouple from this implementation, and then we will be able to have our own naming: https://github.com/mysteriumnetwork/node/issues/854

File: core/service/manager_test.go (hunk 1 above)
Comment (donce) [reply core/service/manager_test.go]: Created an issue for this not to polute this PR with more changes (this PR is already up for 2 days!): https://github.com/mysteriumnetwork/node/issues/854

File: session/manager.go (hunk 3 above)
Comment (donce) [reply session/manager.go]: Oh, now I got what you meant - will fix this.

**Pull request merged**